src/main/java/net/wurstclient/hacks/HealthTagsHack.java

```diff
@@ -7,44 +7,108 @@
  */
 package net.wurstclient.hacks;
 
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.entity.Entity;
 import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.mob.MobEntity;
 import net.minecraft.text.MutableText;
 import net.minecraft.text.Text;
 import net.minecraft.util.Formatting;
 import net.wurstclient.Category;
 import net.wurstclient.SearchTags;
+import net.wurstclient.events.RenderListener;
 import net.wurstclient.hack.Hack;
+import net.wurstclient.settings.CheckboxSetting;
+import net.wurstclient.settings.RoundingPrecisionSetting;
+import net.wurstclient.util.RenderUtils;
 
 @SearchTags({"health tags"})
-public final class HealthTagsHack extends Hack
+public final class HealthTagsHack extends Hack implements RenderListener
 {
+	private final CheckboxSetting mobs = new CheckboxSetting("Mobs",
+		"Displays health tags above mobs also.", false);
+	
+	private final CheckboxSetting showMaxHealth =
+		new CheckboxSetting("Show max health", "Also displays the entity's"
+			+ " maximum health in addition to its current health.", false);
+	
+	private final RoundingPrecisionSetting precision =
+		new RoundingPrecisionSetting("Precision",
+			"Rounds the health value to the given number of decimal places.", 0,
+			0, 3);
+	
 	public HealthTagsHack()
 	{
 		super("HealthTags");
 		setCategory(Category.RENDER);
+		addSetting(mobs);
+		addSetting(showMaxHealth);
+		addSetting(precision);
+	}
+	
+	@Override
+	protected void onEnable()
+	{
+		EVENTS.add(RenderListener.class, this);
+	}
+	
+	@Override
+	protected void onDisable()
+	{
+		EVENTS.remove(RenderListener.class, this);
 	}
 	
-	public Text addHealth(LivingEntity entity, Text nametag)
+	@Override
+	public void onRender(MatrixStack matrixStack, float partialTicks)
+	{
+		if(!mobs.isChecked())
+			return;
+		
+		VertexConsumerProvider.Immediate immediate =
+			MC.getBufferBuilders().getEntityVertexConsumers();
+		
+		for(Entity e : MC.world.getEntities())
+		{
+			if(!(e instanceof MobEntity entity))
+				continue;
+			
+			Text text = addHealth(entity, Text.literal(""));
+			RenderUtils.renderTag(matrixStack, text, entity, immediate,
+				0xffffff, !entity.hasCustomName() ? 0.5 : 1, partialTicks);
+		}
+		
+		immediate.draw();
+	}

```

Comment:
I'm not sure the entity list can actually contain entities outside of render distance. This might already be filtered out by the game itself.

There could be a slight performance improvement if we iterate over all entities in `onUpdate()` and cache the filtered MobEntity list for `onRender()`. Mobs only spawn/despawn in game ticks, and that way the filtering is only done 20 times / second even when you have hundreds of FPS.

But I'm not sure it even matters, as the filtering here is fairly light. I think it's fine to leave this as-is for now.